packages/ws/src/ws/WebSocketShard.ts

```diff
@@ -142,7 +142,9 @@ export class WebSocketShard extends AsyncEventEmitter<WebSocketShardEventsMap> {
 		void this.internalConnect();
 
 		try {
-			await promise;
+			await promise?.catch(({ error }) => {
```

Comment:
This is... very silly, we're in a `try` block and attach a `.catch` to the promise, just use a `catch` block instead.